### ISSUE TITLE
Is joint training used?

### ISSUE THREAD

**Issue by wytcsuch** [HUuxiaobin/Face-Super-Resolution-Guided-by-3D-Facial-Priors#6]
Thank you again for your contribution. I have been studying your paper and code recently. There are three questions:
1. According to your codes, the generation of 3D rendering image and face super resolution are not trained together. But in this paper, I understand that the two branches are joint training, and there is a render loss between rendering face and HR. If my understanding is wrong, please correct me.
2. If, as I understand, the two branches are trained jointly, how do you use tensorflow and pytorch together for training, because tensorflow will occupy all the GPU memory by default whilch make pytorch unable to run?
3. At present, only pixel loss between SR and HR is optimized. Have you ever tried to introduce GAN to help restruct？
